extensions/wikia/Recirculation/RecirculationController.class.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+class RecirculationController extends WikiaController {
+	const DEFAULT_TEMPLATE_ENGINE = WikiaResponse::TEMPLATE_ENGINE_MUSTACHE;
+	const ALLOWED_TYPES = ['popular', 'shares'];
+	const DEFAULT_TYPE = 'popular';
+
+	public function init() {
+		$type = $this->getVal( 'type' );
+		if (in_array($type, self::ALLOWED_TYPES)) {
+			$this->type = $type;
+		} else {
+			$this->type = self::DEFAULT_TYPE;
+		}
+	}
+
+	public function index() {
+		$fandomDataService = FandomDataService::getInstance();
```

Comment:
Looked at a few other extensions and saw something similar to `getInstance`. May be over engineered for this use case though, so I have no issue with `new FandomDataService();`